xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -170,21 +173,33 @@ void CVAAPIContext::SetValidDRMVaDisplayFromRenderNode()
 void CVAAPIContext::SetVaDisplayForSystem()
 {
   auto win_system = g_Windowing.GetWinSystem();
-  if (win_system == WINDOW_SYSTEM_X11)
+  switch (win_system)
```

Comment:
I am wondering if we should move all this to windowing and get rid of this ifdeffery: 
g_windowing.GetVaDisplay(). platforms that support vaapi can implement this. Others won't include this code anyway.